File: AtCoder_Beginners_Selection/ABC088B/main.nim
import strutils, sequtils, algorithm
proc main* =
    let N: int = parseInt(readLine(stdin))
    var input_list: seq[int] = stdin.readLine.split.map(parseInt)

    var Alice_score: int
    var Bob_score: int

    # input check
    assert(1 <= N and N <= 100)
    assert(1 <= len(input_list) and len(input_list) <= N)
    for i in input_list:
        assert(1 <= i and i <= 100)


    # Aliceが一番大きい値をとるために降順ソート
    let sorted_list = sorted(input_list, SortOrder.Descending)

    if len(sorted_list) mod 2 == 0:
        for i in 0 ..< len(sorted_list):
            if i == 0 or i mod 2 == 0:
                Alice_score += sorted_list[i]
                continue
            Bob_score += sorted_list[i]
    else:
        for i in 0 ..< len(sorted_list):
            if i == 0 or i mod 2 == 0 or i == len(sorted_list):
                Alice_score += sorted_list[i]
                continue
            Bob_score += sorted_list[i]
    echo(Alice_score - Bob_score)
when isMainModule:
    main()

File: AtCoder_Beginners_Selection/ABC088B/main.py
def main():
    N: int = int(input()) # N枚のカード
    input_list: list[int] = [int(i) for i in input().split(" ")] #
    Alice_score: int = 0
    Bob_score: int = 0

    # input_check
    assert(1 <= N <= 100) # Nの制約チェック
    assert(1 <= len(input_list) <= N) # iの制約チェック
    for i in input_list: # aの制約チェック O(100)
        assert( 1 <= i <= 100)

    # Aliceが一番大きい数字をとるため降順ソート
    input_list.sort(reverse=True)

    if len(input_list) % 2 == 0: # 配列が偶数個なら互いにとっていける
        for i in range(0, len(input_list)):
            if i == 0 or i % 2 == 0: # Aliceがとるのは偶数の値
                Alice_score += input_list[i]
                continue
            Bob_score += input_list[i]
    else:
        for i in range(0, len(input_list)):
            if i == 0 or i % 2 == 0 or i == len(input_list) - 1: # Aliceがとるのは、0, 要素が偶数の値、最後の値
                Alice_score += input_list[i]
                continue
            Bob_score += input_list[i]
    print(Alice_score - Bob_score)


if __name__ == "__main__":
    main()
